compiler/il/Aliases.cpp

```diff
@@ -440,22 +440,24 @@ OMR::SymbolReference::getUseDefAliasesBV(bool isDirectCall, bool includeGCSafePo
 #ifdef J9_PROJECT_SPECIFIC
          TR::ResolvedMethodSymbol * resolvedMethodSymbol = _symbol->castToResolvedMethodSymbol();
 
-         if (!comp->getOption(TR_EnableHCR))
+         if (resolvedMethodSymbol->getRecognizedMethod() == TR::java_lang_System_arraycopy)
             {
-            switch (resolvedMethodSymbol->getRecognizedMethod())
-               {
-               case TR::java_lang_System_arraycopy:
-                  {
-                  TR_BitVector * aliases = new (aliasRegion) TR_BitVector(bvInitialSize, aliasRegion, growability);
-                  *aliases |= symRefTab->aliasBuilder.arrayElementSymRefs();
-                  if (comp->generateArraylets())
-                     *aliases |= symRefTab->aliasBuilder.arrayletElementSymRefs();
-                  return aliases;
-                  }
+            TR_BitVector * aliases = new (aliasRegion) TR_BitVector(bvInitialSize, aliasRegion, growability);
+            *aliases |= symRefTab->aliasBuilder.arrayElementSymRefs();
+            if (comp->generateArraylets())
+               *aliases |= symRefTab->aliasBuilder.arrayletElementSymRefs();
 
-                  if (resolvedMethodSymbol->isPureFunction())
-                      return NULL;
+            return aliases;
+            }
+
+         static const char *disableHCRGuards = feGetEnv("TR_DisableHCRGuards");
```

Comment:
Is it possible to rename this as "disableHCRAliases" (similar change to env var name) since we are not really talking about guards here ? Unless your intent was to control this behavior using the same env var used elsewhere in the codebase for controlling HCR guards (?)

Reply:
>Unless your intent was to control this behavior using the same env var used elsewhere in the codebase for controlling HCR guards (?)

It's because of this reason. I found there are a few ways to check if HCR is disabled or not in our code. (1) and (2) are essentially the same. I just added (3) in this change. 

(1) Check `comp->getOption(TR_EnableHCR)`
(2) Check if `getHCRMode` is `TR::none`: https://github.com/eclipse/omr/blob/9b3e65e264a66efca05c8ab283a81686ef03c876/compiler/optimizer/Inliner.cpp#L1821
(3) Check env variable `TR_DisableHCRGuards`: https://github.com/eclipse/omr/blob/9b3e65e264a66efca05c8ab283a81686ef03c876/compiler/optimizer/Inliner.cpp#L1814

Reply:
I don't feel that  `TR_DisableHCRGuards` checking in the inliner is checking if HCR is enabled or not. I think it may just be an env var added for someone to specifically disable the creation of inline guards (say as an experiment for whatever reason). If one believes this viewpoint, then they created an env var named for a very specific purpose in the inliner specific to the creation of HCR guards, which probably will not transfer over to this aliasing spot in the code.

Reply:
`TR_DisableHCRGuards` is also used in VP 

https://github.com/eclipse-openj9/openj9/blob/d21c52dcfec41404e6e71a27e45e8806f1cba94e/runtime/compiler/optimizer/J9ValuePropagation.cpp#L110

https://github.com/eclipse-openj9/openj9/blob/d21c52dcfec41404e6e71a27e45e8806f1cba94e/runtime/compiler/optimizer/J9ValuePropagation.cpp#L1440

Reply:
Removed checking on in [b67aca](https://github.com/eclipse/omr/compare/b89433f9a45e74d67b103706398811f7778d9f19..b67acad90b8e6fd1a1f214d6c7598530770aa6bb). Ready for another review